pkg/trace/sampler/score.go

```diff
@@ -26,33 +26,33 @@ func SampleByRate(traceID uint64, rate float64) bool {
 	return true
 }
 
-func capTo1(f float64) float64 {
-	if f > 1 {
-		return 1
-	}
-	return f
-}
-
 // GetSignatureSampleRate gives the sample rate to apply to any signature.
 // For now, only based on count score.
 func (s *Sampler) GetSignatureSampleRate(signature Signature) float64 {
-	return capTo1(s.GetCountScore(signature))
+	return s.loadRate(s.GetCountScore(signature))
 }
 
 // GetAllSignatureSampleRates gives the sample rate to apply to all signatures.
 // For now, only based on count score.
 func (s *Sampler) GetAllSignatureSampleRates() map[Signature]float64 {
 	m := s.GetAllCountScores()
 	for k, v := range m {
-		m[k] = capTo1(v)
+		m[k] = s.loadRate(v)
 	}
 	return m
 }
 
 // GetDefaultSampleRate gives the sample rate to apply to an unknown signature.
 // For now, only based on count score.
 func (s *Sampler) GetDefaultSampleRate() float64 {
-	return capTo1(s.GetDefaultCountScore())
+	return s.loadRate(s.GetDefaultCountScore())
+}
+
+func (s *Sampler) loadRate(rate float64) float64 {
+	if rate >= s.rateThresholdTo1 {
+		rate = 1
```

Comment:
This is confusing, can we just do the usual
```suggestion
		return 1
```